### ISSUE TITLE
Failed to compile ModuleInspector

### ISSUE THREAD

**Issue by wonbyte** [SublimeHaskell/SublimeHaskell#182]
SublimeHaskell: Failed to compile ModuleInspector

C:\Users\Greg\AppData\Roaming\Sublime Text 3\Packages\SublimeHaskell\ModuleInspector.hs:25:18:

    Ambiguous module name `Documentation.Haddock':

      it was found in multiple packages:

      haddock-api-2.15.0 haddock-2.14.3

Using x64 windows 7.

**Comment by kindlychung** [comment]
I am using ghc 7.8.3 and I got this error, too:

    /Library/Application Support/Sublime Text 2/Packages/SublimeHaskell/ModuleInspector.hs:25:18:
    Ambiguous module name ‘Documentation.Haddock’:
    it was found in multiple packages:
    haddock-api-2.15.0.2 haddock-2.14.3

**Comment by ibloom** [comment]
Same here haven't figured it out yet.

**Comment by markogresak** [comment]
@kindlychung, @ibloom have you tried `hsdev` branch, like @geraldus suggested?
You have to:
 - remove SublimeHaskell (if you've installed it via package control)
 - in terminal, go to package folder (`Preferences` -> `Browse Packages...` to find the path)
 - clone SublimeHaskell repository, `cd` into it and checkout `hsdev` branch

**Comment by AndrewBrinker** [comment]
This information should maybe be added to the README. Switching to the hsdev branch totally solves the issue, and finding that out immediately would likely be helpful to other people.

**Comment by mantkiew** [comment]
I am trying this branch today but my Sublime freezes completely "Not responding". I can see; however, that in Task Manager it's CPU and memory usage values are changing. I have the following settings:

hsdev: true
hdevtools: false
ghc-mod: true

I uninstalled SublimeHaskell, added the hsdev branch using "Add Repository" and installed again.

**Comment by kindlychung** [comment]
I have got the same problem as @mantkiew , I am on OS X Yosemite. Sublime version 2.

**Comment by keathley** [comment]
As others have said, I was able to get this working by manually cloning the repo into my sublime packages and checking out the hsdev branch.

Is there a better fix for this or should this information be added to the README?

**Comment by kbpal** [comment]
Regarding the original 'haddock' issue: I've shown how this can be resolved at :  7:52  in : 
http://goo.gl/Xc9Htb
Also addressed are most of the other errors reported in this and other threads.

**Comment by keathley** [comment]
Depending on when #132 is going to be merged into master it may or may not make sense to add the hsdev "fix" to the README.  I'm not sure how much work needs to be done to finish that PR.

**Comment by mantkiew** [comment]
The video linked to by @kbpal shows how absolutely miserable the user's life is. That is why I worked out the specific versions and creted the pull request #214. At least on Windows, it get you started quickly.

**Comment by mooreniemi** [comment]
for future users to copy pasta:
`git clone https://github.com/SublimeHaskell/SublimeHaskell.git` into the `/Packages` dir
`git checkout hsdev`
and my woes were over

**Comment by alfonsoperez** [comment]
Thanks so much guys!!

**Comment by ulidtko** [comment]
@mooreniemi FWIW, git can do even better:

    git clone --branch hsdev https://github.com/SublimeHaskell/SublimeHaskell.git

**Comment by dogweather** [comment]
:+1: Thanks @ulidtko

**Comment by Jipperism** [comment]
After a number of tries I'm using the latest version of the haskell platform, and the branch specified here with ST3. It seems to work, however, I'm getting the following error message. Does this appear in other people's ST3 console as well or do I need to look into it more? I'm a bit at a loss as to how to fix this since all threads seem to point here so please tell me if this is the wrong place to ask.

```
 Traceback (most recent call last):
  File "/opt/sublime_text/sublime_plugin.py", line 74, in reload_plugin
    m = importlib.import_module(modulename)
  File "./importlib/__init__.py", line 90, in import_module
  File "<frozen importlib._bootstrap>", line 1584, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 584, in _check_name_wrapper
  File "<frozen importlib._bootstrap>", line 1022, in load_module
  File "<frozen importlib._bootstrap>", line 1003, in load_module
  File "<frozen importlib._bootstrap>", line 560, in module_for_loader_wrapper
  File "<frozen importlib._bootstrap>", line 868, in _load_module
  File "<frozen importlib._bootstrap>", line 313, in _call_with_frames_removed
  File "/home/jip/.config/sublime-text-3/Packages/SublimeHaskell/repl.py", line 17, in <module>
    import SublimeREPL.sublimerepl as sublimerepl
ImportError: No module named 'SublimeREPL'
reloading plugin SublimeHaskell.stylishhaskell
reloading plugin SublimeHaskell.sublime_haskell_common
reloading plugin SublimeHaskell.symbols
reloading plugin SublimeHaskell.util
reloading plugin Terminal.Terminal
reloading plugin User.toggle_vintageous
plugins loaded
Package Control: Skipping automatic upgrade, last run at 2016-01-15 21:40:00, next run at 2016-01-15 22:40:00 or after
Sublime Haskell: drop prepared completions
```

**Comment by mvoidex** [comment]
@Jipperism, are you using `hsdev` branch? Do you have SublimeREPL plugin installed?

**Comment by Jipperism** [comment]
Yes i am using hsdev branch, I followed the instructions for installing it through the sublime package manager by adding the repository. I figured out I did not have SublimeREPL installed indeed, and installing it fixed the error messages at startup so my question is probably not in the right place anymore since I do not get the 'Failed to compile moduleinspector' error anymore. None of the features is working tho, it seems to crash on trying to autocomplete and i cannot insert or view types. Should I start my own thread?
